lib/spack/spack/test/stage.py

```diff
@@ -93,94 +168,220 @@ def get_stage_path(stage, stage_name):
         return stage.path
 
 
-@pytest.fixture()
-def tmpdir_for_stage(mock_archive):
-    """Uses a temporary directory for staging"""
-    current = spack.paths.stage_path
+@pytest.fixture
```

Comment:
Ok this is kind of cool.  Didn't know [`pytest.fixture`](https://github.com/pytest-dev/pytest/blob/2e04771893f17b571e6ab48f6884a6a70cde8fc8/_pytest/python.py#L132) was written to work with or without parens.  I checked the version of `pytest` that we vendor (it's old for 2.6 compatibility) and seems to work there too.  Cool.